src/nacl/utils.py

```diff
@@ -14,57 +14,63 @@
 
 
 import os
-from typing import SupportsBytes
+from typing import SupportsBytes, Type, TypeVar
 
 import nacl.bindings
 from nacl import encoding
 
+C = TypeVar("C", bound="EncryptedMessage")
+
 
 class EncryptedMessage(bytes):
     """
     A bytes subclass that holds a messaged that has been encrypted by a
     :class:`SecretBox`.
     """
 
-    _nonce = object
-    _ciphertext = object
+    _nonce: bytes
+    _ciphertext: bytes
 
     @classmethod
-    def _from_parts(cls, nonce, ciphertext, combined):
+    def _from_parts(
+        cls: Type[C], nonce: bytes, ciphertext: bytes, combined: bytes
```

Comment:
Can we just pass cls: "EncryptedMessage" as a string here?

Reply:
I'm fine either way, but @Dreamsorcerer seemed pretty keen on this in #692. Perhaps they could advise?

Reply:
Problem is that the code is not returning `EncryptedMessage()`, it's returning `cls()`. If you have a subclass, the return type is not `EncryptedMessage`, hence the need for a `TypeVar`.

I'd probably call it `_EncryptedMessage` or something though, to avoid any chance of someone accidentally reusing the type when it is bound to `EncryptedMessage`.

Reply:
I’m not overly concerned about supporting subclassing of this type (don’t subclass things, this shouldn’t even be subclassing bytes, sigh), but I’m okay if we want to do it this way. Let’s do the private prefix if we keep it this way though.

Reply:
> I’m not overly concerned about supporting subclassing of this type.

Sure, but my argument is that it should match the code. If we changed the return type to `EncryptedMessage`, then I expect to see the implementation use `EncryptedMessage`, rather than `cls`.

Reply:
The `Self` type proposed in [PEP 673](https://www.python.org/dev/peps/pep-0673/) would make this neater.